### PR TITLE
refactor: convert config-validator to TypeScript

### PR DESCRIPTION
**Pull request by rarkins**
BREAKING CHANGE: path for the config validator is now in lib/ and dist/ and not bin/

### CODE REVIEW

**Comment by rarkins** [comment]
@viceice have I done something wrong? 
image

**Comment by viceice** [comment]
🤔 will have a look later

**Comment by rarkins** [comment]
I think I need to update the package.json script

**Comment by viceice** [comment]
I'm not sure what happend, should just work

File: lib/config-validator.ts [hunk 1]
```diff
@@ -1,16 +1,20 @@
-#!/usr/bin/env node
-
-const fs = require('fs-extra');
-const { validateConfig } = require('../dist/config/validation');
-const { massageConfig } = require('../dist/config/massage');
-const { getConfig } = require('../dist/config/file');
-const { configFileNames } = require('../dist/config/app-strings');
+// istanbul ignore file
```
Comment (viceice) [inline lib/config-validator.ts:1]:
```suggestion
#!/usr/bin/env node
// istanbul ignore file
```

**Pull request merged**